src/Core/View.js

```diff
@@ -609,9 +609,10 @@ View.prototype.pickObjectsAt = function pickObjectsAt(mouseOrEvt, ...where) {
 
             // does this layer have a custom picking function?
             if (layer.pickObjectsAt) {
-                results.splice(
-                    results.length, 0,
-                    ...layer.pickObjectsAt(this, mouse));
+                const sp = layer.pickObjectsAt(this, mouse);
+                for (let i = 0; i < sp.length; i++) {
+                    results.push(sp[i]);
```

Comment:
The problem was `javascript maximum call stack size exceeded` because the array might be very large.
Your suggestion unfortunately won't work either (`Do not use this method if the second array`) but I'll add a comment to explain why we need to do this.